### PR TITLE
add ipam driver none support

### PR DESCRIPTION
**Pull request by Luap99**
Allow creating interfaces only with no ip addresses assigned.
Also combine some duplicated code for bridge/macvlan.

Ref https://github.com/containers/common/pull/967

### CODE REVIEW

File: src/network/core_utils.rs [hunk 1]
```diff
@@ -25,6 +25,129 @@ pub struct CoreUtils {
...
+    pub gateway_addresses: Vec<ipnet::IpNet>,
+    pub ipv6_enabled: bool,
+    // result for podman
+    pub net_addresses: Vec<types::NetAddress>,
+    pub nameservers: Vec<IpAddr>,
+}
+
+const IPAM_HOSTLOCAL: &str = "host-local";
+const IPAM_DHCP: &str = "dhcp";
+const IPAM_NONE: &str = "none";
```
Comment (flouthoc) [inline src/network/core_utils.rs:40]: Not a blocker: Should we move these constants to https://github.com/containers/netavark/blob/main/src/network/constants.rs as they might get used by other parts of code in future.

File: src/network/core_utils.rs [hunk 2]
```diff
@@ -25,6 +25,129 @@ pub struct CoreUtils {
     pub networkns: String,
 }
 
+pub struct IPAMAddresses {
```
Comment (flouthoc) [inline src/network/core_utils.rs:28]: This can be moved to https://github.com/containers/netavark/blob/main/src/network/internal_types.rs

**Review by flouthoc — APPROVED** [review]
LGTM, just two nits above.

File: src/network/core_utils.rs (hunk 2 above)
Comment (Luap99) [reply src/network/core_utils.rs:28]: yeah good idea

File: src/network/core_utils.rs (hunk 1 above)
Comment (Luap99) [reply src/network/core_utils.rs]: done

**Comment by Luap99** [comment]
@flouthoc @mheon PTAL

**Review by flouthoc — APPROVED** [review]
LGTM

/lgtm
/approve

**Pull request merged**